src/pal/inc/rt/palrt.h

```diff
@@ -1502,6 +1502,7 @@ typedef struct _DISPATCHER_CONTEXT {
     PRUNTIME_FUNCTION FunctionEntry;
     DWORD EstablisherFrame;
     DWORD TargetIp;
+    PCONTEXT CurrentRecord;
```

Comment:
It seems the parent function of the filter (the one that contains it) always has EBP frame, otherwise the filter funclet code would not be able to access the locals of the parent. If that's the case, we could just rely on the fact that the EBP is stored at the callerContext->SP - sizeof(TADDR) and we could just modify the GetFrameRestoreBase to return GetCallerSp(pContextRecord) - 2 * sizeof(TADDR) like we already do at several other places for EBP based frames and leave the rest of the code unchanged.